### PR TITLE
Support Different Number of Max Children

### PR DESCRIPTION
**Pull request by nibanks**
Add support for setting child max, for platforms that define a different maximum number of supported children. 

If the maximum number of children is higher than expected, the cert test fails because all children are able to join.

### CODE REVIEW

File: src/ncp/spinel.h [hunk 1]
```diff
@@ -532,7 +532,12 @@ typedef enum
     /** Format: `A(T(ESLCcCbLL))`
      *  eui64, rloc16, age, inLqi ,aveRSS, mode, isChild. linkFrameCounter, mleCounter
      */
-    SPINEL_PROP_THREAD_NEIGHBOR_TABLE  = SPINEL_PROP_THREAD_EXT__BEGIN + 11,
+    SPINEL_PROP_THREAD_NEIGHBOR_TABLE   = SPINEL_PROP_THREAD_EXT__BEGIN + 11,
```
Comment (abtink) [inline src/ncp/spinel.h]: I wonder why adding a white-space before `=` here?  
I think this makes it mis-aligned with the rest of the prop definitions.

File: src/ncp/spinel.h [hunk 2]
```diff
@@ -532,7 +532,12 @@ typedef enum
...
+
+    /// Thread Max Child Count
+    /** Format: `C`
+     */
+    SPINEL_PROP_THREAD_CHILD_COUNT_MAX  = SPINEL_PROP_THREAD_EXT__BEGIN + 12,
```
Comment (abtink) [inline src/ncp/spinel.h]: Same here, the `=` does not look aligned with rest.

File: src/ncp/spinel.h (hunk 1 above)
Comment (nibanks) [inline src/ncp/spinel.h]: I just looked, and it looks like the last several have alignment issues.

Comment (nibanks) [inline src/ncp/spinel.h]: Submitted a fix.

Comment (abtink) [inline src/ncp/spinel.h]: Yes. Indeed. The `SPINEL_PROP_THREAD_ROUTER_SELECTION_JITTER` and `SPINEL_PROP_THREAD_PREFERRED_ROUTER_ID` are not aligned correctly.
But the `SPINEL_PROP_THREAD_NEIGHBOR_TABLE` seems correct. So let's keep it as is in this commit/PR. For the other two I will submit a separate (style only) commit/PR.

**Pull request merged**